test/jdk/sun/security/pkcs11/ec/ReadCertificates.java

```diff
@@ -182,6 +190,15 @@ public void main(Provider p) throws Exception {
                 }
             } catch (SignatureException | InvalidKeyException e) {
                 System.out.println("OK: " + e);
+            } catch (ProviderException pe) {
+                if (pe.getMessage().contains("cancel failed")
+                    && "s390x".equals(System.getProperty("os.arch"))
+                ) {
+                    System.out.println("NSS error code is different in s390x.");
+                    pe.printStackTrace();
+                    return;
```

Comment:
This bypasses the code that prints "OK"; is that not necessary for the test to pass?

Reply:
Maybe we want `break` instead of `return`?

Reply:
Yes, you are right. But we don't want to break. Rather move on to the next iteration. So, I updated accordingly.